pyvisfile/xdmf/__init__.py

```diff
@@ -0,0 +1,1119 @@
+            name: Optional[str] = None,
+            itype: Optional[DataItemType] = DataItemType.Uniform,
+            ntype: Optional[DataItemNumberType] = DataItemNumberType.Float,
+            precision: Optional[int] = 4,
+            reference: Optional[str] = None,
+            function: Optional[str] = None,
+            endian: Optional[DataItemEndian] = DataItemEndian.Native,
+            dformat: Optional[DataItemFormat] = DataItemFormat.XML,
+            parent: Optional[Element] = None,
+            data: Optional[str] = None,
+            ):
+        """
+        :param parent: if provided, *self* is appended to the element.
+        :param reference: name of another :class:`DataItem`.
+        :param data: data contained inside the :class:`DataItem`. This is
+            usually a path to a binary file.
+        """
+
+        self._dimensions = dimensions
+
+        super().__init__(parent, "DataItem", {
+            "Name": name,
+            "ItemType": itype.name if itype is not None else itype,
+            "Dimensions": dimensions,
+            "NumberType": ntype.name if ntype is not None else ntype,
+            "Precision": precision,
+            "Reference": reference,
+            "Function": function,
+            "Endian": endian.name if endian is not None else endian,
+            "Format": dformat.name if dformat is not None else dformat,
+            })
+
+        if data is not None:
+            self.text = data
+
+    @property
+    def dimensions(self):
+        if self._dimensions is None:
+            raise AttributeError
+
+        return self._dimensions
+
+    @classmethod
+    def as_reference(cls, reference: str, *,
+            parent: Optional[Element] = None) -> "DataItem":
+        if not reference.startswith("/"):
+            reference = f"/Xdmf/Domain/DataItem[@Name='{reference}']"
+
+        return cls(
+                reference="XML",
+                data=reference,
+                itype=None, ntype=None, precision=None,
+                endian=None, dformat=None,
+                parent=parent)
+
+
+def data_item_from_numpy(
+        ary: np.ndarray, *,
+        name: Optional[str] = None,
+        parent: Optional[Element] = None,
+        data: Optional[str] = None) -> DataItem:
+    """Create a :class:`DataItem` from a given :class:`~numpy.ndarray`."""
+
+    if data is None:
+        dformat = DataItemFormat.XML
+    else:
+        dformat = _data_item_format_from_str(data)
+
+    return DataItem(
+            name=name,
+            dimensions=ary.shape,
+            itype=DataItemType.Uniform,
+            ntype=_numpy_to_xdmf_number_type(ary.dtype),
+            precision=ary.dtype.itemsize,
+            endian=_system_to_xdmf_endian(),
+            dformat=dformat,
+            parent=parent,
+            data=data,
+            )
+
+
+def join_data_items(
+        items: Tuple[DataItem, ...], *,
+        parent: Optional[Element] = None) -> DataItem:
```

Comment:
What are the semantics of "joined" data items (aka docstring request)?

Reply:
Added some docs in b895952, but also removed it from the public docs.

The "join" was meant in the sense of using the XDMF [Function](https://www.xdmf.org/index.php/XDMF_Model_and_Format#Function) type, which has a `JOIN` thing.